setup more detailed mkdocs documentation (#880)
pypa/setuptools_scm: README.md

# setuptools_scm
[![github ci](https://github.com/pypa/setuptools_scm/workflows/python%20tests+artifacts+release/badge.svg)](https://github.com/pypa/setuptools_scm/actions)
[![tidelift](https://tidelift.com/badges/package/pypi/setuptools-scm) ](https://tidelift.com/subscription/pkg/pypi-setuptools-scm?utm_source=pypi-setuptools-scm&utm_medium=readme)

## about

[setuptools-scm] extracts Python package versions from `git` or
`hg` metadata instead of declaring them as the version argument

> or in a SCM managed file.

Reviewer: ```suggestion
or in an SCM managed file.
```

Additionally, [setuptools-scm] provides setuptools
with a list of files that are managed by the SCM <br/>
(i.e. it automatically adds **all of** the SCM-managed files to the sdist).<br/>

> Unwanted files must be excluded by discarding them via `MANIFEST.in`.

Reviewer: ```suggestion
Unwanted files must be excluded via `MANIFEST.in`.
```

## `pyproject.toml` usage

The preferred way to configure [setuptools-scm] is to author
settings in a `tool.setuptools_scm` section of `pyproject.toml`.

> This feature requires Setuptools 60 or later

Reviewer: ```suggestion
This feature requires setuptools 60 or later.
```

First, ensure that [setuptools-scm] is present during the project's

> built step by specifying it as one of the build requirements.

Reviewer: ```suggestion
build step by specifying it as one of the build requirements.
```

```toml
[build-system]
requires = [
    "setuptools>=60",
    "setuptools-scm>=8.0"]
```

That will be sufficient to require [setuptools-scm] for projects
that support [PEP-518] like [pip] and [build].

[pip]: https://pypi.org/project/pip
[build]: https://pypi.org/project/build

> [PEP-518]: https://www.python.org/dev/peps/pep-0518/

Reviewer: ```suggestion
that support [PEP 518] like [pip] and [build].

[pip]: https://pypi.org/project/pip
[build]: https://pypi.org/project/build
[PEP 518]: https://peps.python.org/pep-0518/
```

To enable version inference, you need to set the version
dynamically in the `project` section of `pyproject.toml`:

```toml title="pyproject.toml"
[project]
# version = "0.0.1"  # Remove any existing version parameter.
dynamic = ["version"]
[tool.setuptools_scm]
```

> Additionally, a version file can be written by specifying

Reviewer: ```suggestion
Additionally, a version file can be written by specifying:
```

```toml title="pyproject.toml"
[tool.setuptools_scm]
version_file = "pkg/_version.py"
```

Where `pkg` is the name of your package.

If you need to confirm which version string is being generated or debug the configuration,
you can install [setuptools-scm] directly in your working environment and run:

[setuptools-scm]: https://github.com/pypa/setuptools_scm

> ```commandline

Reviewer: ```suggestion
```console
```

$ python -m setuptools_scm
# To explore other options, try:
$ python -m setuptools_scm --help
```

## Interaction with Enterprise Distributions

> Some enterprise distributions like RHEL7 and others

Reviewer: ```suggestion
Some enterprise distributions like RHEL7
```

> ship rather old setuptools versions due to various release management details.

Reviewer: Can remove this vague reason:
```suggestion
ship rather old setuptools versions.
```

> In those case its typically possible to build by using a sdist against `setuptools_scm<2.0`.

Reviewer: ```suggestion
In those cases its typically possible to build by using an sdist against `setuptools_scm<2.0`.
```

As those old setuptools versions lack sensible types for versions,
modern [setuptools-scm] is unable to support them sensibly.

> It's strongly recommended to build a wheel artifact using modern python and setuptools,

Reviewer: ```suggestion
It's strongly recommended to build a wheel artifact using modern Python and setuptools,
```
